lib/metadataTypes/Folder.js

```diff
@@ -451,25 +449,28 @@ class Folder extends MetadataType {
 
     /**
      * Helper to retrieve the folders as promise
-     * @param {String[]} [overrideFields] Returns specified fields even if their retrieve definition is not set to true
+     * @param {String[]} [additionalFields] Returns specified fields even if their retrieve definition is not set to true
      * @param {Boolean} [queryAllAccounts] which queryAllAccounts setting to use
      * @returns {Promise<Object>} soap object
      */
-    static async retrieveHelper(overrideFields, queryAllAccounts) {
+    static async retrieveHelper(additionalFields, queryAllAccounts) {
         const options = { queryAllAccounts: !!queryAllAccounts };
         let status = 'MoreDataAvailable';
         const Results = [];
         do {
-            const response = await new Promise((resolve, reject) => {
-                // filtered out path as we need them to be stored locally, but do not want to try and retrieve
-                this.client.SoapClient.retrieve(
-                    'DataFolder',
-                    this.getFieldNamesToRetrieve(overrideFields).filter(
-                        (field) => !field.includes('Path')
-                    ),
-                    options,
-                    (ex, response) => (ex ? reject(ex) : resolve(response))
-                );
+            let response;
```

Comment:
ditto

Reply:
here it makes sense to keep it as-is because `response` needs to be available outside the await (lines 462ff) because we use it on 476